src/testsuite/arith128_test_i64.c

```diff
@@ -1094,6 +1272,758 @@ test_muloud (void)
+  e = (vui128_t) CONST_VINT128_DW128 ( 0x0UL, 0xfffffffe00000001UL );
+  k = vec_muleud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuleud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuleud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t) CONST_VINT64_DW ( 0xffffffff00000000UL, 0xffffffff00000000UL );
+  j = (vui64_t) CONST_VINT64_DW ( 0xffffffff00000000UL, 0xffffffff00000000UL );
+  e = (vui128_t) CONST_VINT128_DW128 ( 0xfffffffe00000001UL, 0x0UL );
+  k = vec_vmuleud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuleud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuleud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t) CONST_VINT64_DW ( 0xffffffffffffffffUL, 0x0000000000000000UL );
+  j = (vui64_t) CONST_VINT64_DW ( 0xffffffffffffffffUL, 0x0000000000000000UL );
+  e = (vui128_t) CONST_VINT128_DW128 (0xfffffffffffffffeUL, 0x0000000000000001UL );
+  k = vec_vmuleud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuleud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuleud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t) CONST_VINT64_DW ( 2000000000000000000UL, 1000000000000000000UL );
+  j = (vui64_t) CONST_VINT64_DW ( 1UL, -1UL );
+  e = (vui128_t) CONST_VINT128_DW128 ( 0UL, 2000000000000000000UL );
+  k = vec_vmuleud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuleud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuleud:", (vui128_t)k, (vui128_t) e);
+
+  return (rc);
+}
+
+int
+test_vmuloud (void)
+{
+  vui64_t i, j;
+  vui128_t k, e;
+  int rc = 0;
+
+  printf ("\ntest_vmuloud Vector Multiply Odd Unsigned doublewords\n");
+  i = (vui64_t) CONST_VINT64_DW ( 1, 2 );
+  j = (vui64_t) CONST_VINT64_DW ( 101, 102 );
+  e = (vui128_t) CONST_VINT128_DW128 (0, 204);
+  k = vec_vmuloud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuloud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuloud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t) CONST_VINT64_DW ( 1000000000, 1000000000 );
+  j = (vui64_t) CONST_VINT64_DW ( 1000000000, 1000000000 );
+  e = (vui128_t) CONST_VINT128_DW128 ( 0, 1000000000000000000UL );
+  k = vec_vmuloud(i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2int64 ("vmuloud ( ", i);
+  print_v2int64 ("         ,", j);
+  print_vint128 ("        )=", (vui128_t)k);
+#endif
+  rc += check_vuint128x ("vec_vmuloud:", (vui128_t)k, (vui128_t) e);
+
+  i = (vui64_t) CONST_VINT64_DW ( 10000000000, 1000000000 );
+  j = (vui64_t) CONST_VINT64_DW ( 10000000000, 1000000000 );
+  e = (vui128_t) CONST_VINT128_DW128 ( 0x0, 0x0de0b6b3a7640000UL );
```

Comment:
magic number.  I think it would be better to express `e` in decimal as well (`1000000000000000000UL`, or should it be `ULL`?)

Reply:
This trick is final const is an __int128 and not ULL. The compiler does noit support __int128 const!
I will work on macros to improve the generation of __int128 constants from a pair of ULL consts.

Later